x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/fulltext/QueryStringFunctionTests.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.fulltext;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.AbstractFunctionTestCase;
+import org.elasticsearch.xpack.esql.expression.function.FunctionName;
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+import org.hamcrest.Matcher;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.equalTo;
+
+@FunctionName("qstr")
```

Comment:
Thanks for including the generated docs.

Don't forget to commit the changes to `AbstractFunctionTestCase`, otherwise you may get these messages the next time `./gradlew -p x-pack/plugin/esql/ test` runs.

```
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    docs/reference/esql/functions/description/qstr.asciidoc
	deleted:    docs/reference/esql/functions/examples/qstr.asciidoc
	deleted:    docs/reference/esql/functions/kibana/definition/qstr.json
	deleted:    docs/reference/esql/functions/kibana/docs/qstr.md
	deleted:    docs/reference/esql/functions/layout/qstr.asciidoc
	deleted:    docs/reference/esql/functions/parameters/qstr.asciidoc
	deleted:    docs/reference/esql/functions/signature/qstr.svg
	deleted:    docs/reference/esql/functions/types/qstr.asciidoc
```

Reply:
I did that but reverted the change, as this implies changes from the Categorize function. I've opened [this PR](https://github.com/elastic/elasticsearch/pull/113080) for dealing with that specific change. Can you please review?

Reply:
I guess we are pending on the PR to change `AbstractFunctionTestCase` to generate docs for snapshot functions, without it if just checking in `QSTR` related docs, it will be a surprise the next time people run `./gradlew -p x-pack/plugin/esql/ test` before pushing PRs, these docs will be marked as deleted.

Reply:
I just merged https://github.com/elastic/elasticsearch/pull/113080 👍